repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -41,7 +47,34 @@ class Interpreter(shellPrompt0: Ref[String],
         printer
       )
     } finally Thread.currentThread().setContextClassLoader(oldClassloader)
-  } yield out
+  }
+
+  def processScript(code: String): Unit = {
+    val blocks = Parsers.splitScript(code).map(preprocess(_, eval.getCurrentLine))
+    val errors = blocks.collect{ case Res.Failure(err) => err }
+    if(!errors.isEmpty) 
+      stdout(Console.RED + errors.mkString("\n") + Console.RESET + "\n")
+    else{
+      loop(blocks.collect{ case Res.Success(o) => o })
+    }
+
+    @tailrec def loop(blocks: Seq[Preprocessor.Output]): Unit = {
+      if(!blocks.isEmpty){
+        val Preprocessor.Output(code, _) = blocks.head //pretty printing results is disabled for scripts
+        val ev = evaluateLine(code, Seq(), _ => ())
+        ev match {
```

Comment:
Can't you combine this match statement with the one below? You may have to specify `loop` twice, but having two identical matches looks very awkward